src/xdocs/config_filters.xml

```diff
@@ -152,35 +152,35 @@
               <tr>
                   <td>offCommentFormat</td>
                   <td>Specify comment pattern to trigger filter to begin suppression.</td>
-                  <td><a href="property_types.html#regexp">Regular Expression</a></td>
+                  <td><a href="property_types.html#Pattern">Pattern</a></td>
                   <td><code>"CHECKSTYLE:OFF"</code></td>
                   <td>3.5</td>
               </tr>
               <tr>
                   <td>onCommentFormat</td>
                   <td>Specify comment pattern to trigger filter to end suppression.</td>
-                  <td><a href="property_types.html#regexp">Regular Expression</a></td>
+                  <td><a href="property_types.html#Pattern">Pattern</a></td>
                   <td><code>"CHECKSTYLE:ON"</code></td>
                   <td>3.5</td>
               </tr>
               <tr>
                   <td>checkFormat</td>
                   <td>Specify check pattern to suppress.</td>
-                  <td><a href="property_types.html#regexp">Regular Expression</a></td>
+                  <td><a href="property_types.html#Pattern">Pattern</a></td>
                   <td><code>".*"</code></td>
                   <td>3.5</td>
               </tr>
               <tr>
                   <td>messageFormat</td>
                   <td>Specify message pattern to suppress.</td>
-                  <td><a href="property_types.html#regexp">Regular Expression</a></td>
+                  <td><a href="property_types.html#Pattern">Pattern</a></td>
                   <td><code>null</code></td>
                   <td>3.5</td>
               </tr>
               <tr>
                   <td>idFormat</td>
                   <td>Specify check ID pattern to suppress.</td>
-                  <td><a href="property_types.html#regexp">Regular Expression</a></td>
+                  <td><a href="property_types.html#Pattern">Pattern</a></td>
```

Comment:
it is String, please keep it `Regular Expression`, if setter is not `Pattern` it should be abstract non-existing type "Regular Expression" that user cannot instantiate of.

Apply the same to all other.

Reply:
Done